### PR TITLE
Multi-versioned extensions: propagate version info in the binaries

### PR DESCRIPTION
**Pull request by dvoytenko**
Partial for #33001.

For multi-version support we need to know what version the extension is being registered and whether or not it's the "latest" version.

### CODE REVIEW

File: build-system/tasks/extension-helpers.js [hunk 1]
```diff
@@ -523,22 +523,30 @@ function buildExtensionCss(path, name, version, options) {
...
   await compileJs(
     path + '/',
     filename,
     './dist/v0',
     Object.assign(options, {
       toName: `${name}-${version}.max.js`,
       minifiedName: `${name}-${version}.js`,
-      latestName: version === latestVersion ? `${name}-latest.js` : '',
+      latestName: latest ? `${name}-latest.js` : '',
+      esmPassCompilation: argv.esm || argv.sxg || false,
```
Comment (jridgewell) [inline build-system/tasks/extension-helpers.js:535]: I believe I removed this from master.
```suggestion
```

**Review by jridgewell — CHANGES_REQUESTED** [review]
How is this going to be used?

**Comment by dvoytenko** [comment]
@jridgewell 

> How is this going to be used?

To separate extension versions from each other the Runtime will need to know to which version the registrations belong. This code is currently [here](https://github.com/ampproject/amphtml/blob/81b8b9bbbdd272ebd6db43f72533dc0725ac4783/src/service/extensions-impl.js#L527). In the future it'd need to know both the `extensionId` and the `version`. 

The `latest` is a bit trickier. A version can be requested by its number of "latest". So we need to know when they are one and the same.

File: build-system/tasks/extension-helpers.js (hunk 1 above)
Comment (dvoytenko) [reply build-system/tasks/extension-helpers.js]: Ouch. Bad merge. Removed.

**Review by jridgewell — APPROVED** [review]

File: build-system/compile/compile-wrappers.js [hunk 2]
```diff
@@ -71,7 +73,8 @@ exports.extension = function (
   // Use a numeric value instead of boolean. "m" stands for "module"
   const m = isModule ? 1 : 0;
   return (
-    `(self.AMP=self.AMP||[]).push({n:"${name}",${priority}${deps}` +
+    `(self.AMP=self.AMP||[]).push({n:"${name}",ev:"${version}",l:${latest},` +
```
Comment (erwinmombay) [inline build-system/compile/compile-wrappers.js:76]: qq did we add these new fields to amp.extern.js?

Comment (erwinmombay) [reply build-system/compile/compile-wrappers.js:76]: to @dvoytenko

Comment (dvoytenko) [reply build-system/compile/compile-wrappers.js:76]: Thanks for the reminder. PTAL: https://github.com/ampproject/amphtml/pull/33098

**Pull request merged**